futurex_openedx_extensions/helpers/management/commands/course_access_roles_clean_up.py

```diff
@@ -0,0 +1,146 @@
+
+import copy
+from typing import Dict
+
+from common.djangoapps.student.models import CourseAccessRole
+from django.contrib.auth import get_user_model
+from django.core.management import BaseCommand, CommandParser
+
+from futurex_openedx_extensions.dashboard.serializers import UserRolesSerializer
+from futurex_openedx_extensions.helpers import constants as cs
+from futurex_openedx_extensions.helpers.exceptions import FXExceptionCodes
+from futurex_openedx_extensions.helpers.roles import (
+    cache_refresh_course_access_roles,
+    get_user_course_access_roles,
+    update_course_access_roles,
+)
+from futurex_openedx_extensions.helpers.tenants import get_all_tenant_ids, get_course_org_filter_list
+from futurex_openedx_extensions.helpers.users import is_system_staff_user
+
+
+class Command(BaseCommand):
+    """
+    Creates enrollment codes for courses.
+    """
+
+    help = 'Cleans up course access roles for all users'
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        """Add arguments to the command."""
+        parser.add_argument(
+            '--commit',
+            action='store',
+            dest='commit',
+            default='no',
+            help='Commit changes, default is no (just perform a dry-run).',
+            type=str,
+        )
+
+    def handle(self, *args: list, **options: Dict[str, str]) -> None:
+        """Handle the command."""
+        commit = (str(options['commit']).lower() == 'yes')
+
+        tenant_ids = get_all_tenant_ids()
+        user_ids = CourseAccessRole.objects.values_list('user_id', flat=True).distinct()
+        user_ids_to_clean = []
+
+        print('-' * 80)
+        print(f'{len(user_ids)} users to process..')
+        for user_id in user_ids:
+            cache_refresh_course_access_roles(user_id)
+            roles = get_user_course_access_roles(user_id)
+            if roles['useless_entries_exist']:
+                user_ids_to_clean.append(user_id)
+        if not user_ids_to_clean:
+            print('No dirty entries found..')
+        else:
+            print(f'Found {len(user_ids_to_clean)} users with dirty entries..')
+
+        superuser = get_user_model().objects.filter(is_superuser=True).first()
+        all_orgs = get_course_org_filter_list(tenant_ids)['course_org_filter_list']
+        fx_permission_info = {
+            'view_allowed_any_access_orgs': all_orgs,
+            'view_allowed_tenant_ids_any_access': tenant_ids,
+        }
+        fake_request = type('Request', (object,), {
+            'fx_permission_info': fx_permission_info,
+            'query_params': {},
+        })
+
+        for user_id in user_ids_to_clean:
+            user = get_user_model().objects.get(id=user_id)
+            print(f'\nCleaning up user {user_id}:{user.username}:{user.email}...')
+            if not user.is_active:
+                print('**** User is not active, skipping..')
+                continue
+            if is_system_staff_user(user):
```

Comment:
Is staff should lose all of its roles to avoid needless permissions appearing on the admin pages.